source/agora/consensus/PreImage.d

```diff
@@ -487,3 +490,73 @@ unittest
     const number_of_cycles = 12;
     testPreImageCycle(cycle_length, number_of_cycles);
 }
+
+///
+version (unittest) public Hash getSeedAt(in Scalar secret, in ulong total_images,
+    in uint cycle_length, in Height at) nothrow
+{
+    auto cycle = PreImageCycle(secret, cycle_length, total_images, Height(0));
+    return cycle[at];
+}
+
+/// To get the cycle seeds for `PreImageCycle`s of some of well-known keys,
+/// which are used for getting 1000 number of pre-images
+version (unittest) public void getCycleSeed (in KeyPair pair, out Hash seed,
```

Comment:
Since this function is only used in `test/Base.d`, move it there and make it private ?

Reply:
It could be. I'll try it.

Reply:
Updated

Reply:
Updated.